### PR TITLE
Replace SnapshotExpectationValue with SaveExpectationValue

### LINKED ISSUE #5743: AerPauliExpectation.compute_variance returns zero variance
### What is the expected enhancement?

AerPauliExpectation.compute_variance returns zero variance as the comment said.
https://qiskit.org/documentation/stubs/qiskit.aqua.operators.expectations.AerPauliExpectation.compute_variance.html

However, AerPauliExpectation has a finite (=non-zero) variance in general. In `SnapshotExpectationValue` used here,  an optional argument of variance is `True`. It seems that this returns variance computed in Aer.

### LINKED ISSUE #6504: Handle Aer deprecation on AerPauliExpectation
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
AerPauliExpectation shows the following deprecate msg:
```
PendingDeprecationWarning: The `SnapshotExpectationValue` instruction will be deprecated in the future. It has been superseded by the `SaveExpectationValue` and `SaveExpectationValueVariance` instructions.
```

### What is the expected behavior?
SaveExpectationValue / SaveExpectationValueVariance should be used instead.

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Resolve #6504

### Details and comments

SaveExpectationValue does not support Non-hermitian operator now. The test is skipped until https://github.com/Qiskit/qiskit-aer/issues/1271.
If this is necessary, don't merge this PR until it is supported.

I cannot resolve 5743 because current design of CircuitSampler does not allow such an extension.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
@ikkoham Thanks for fixing this. Should we hold off merging this until we get feedback on the Aer issue you mention in the unit test.

**Comment by woodsp-ibm** [comment]
@ikkoham It seems that only supporting Hermitian operators is intentional, as per the conversation in the issue you raised on Aer as linked above. As such I think think the best course of action is just simply to remove the unit test with the non hermitian operator as its no longer valid.

**Comment by ikkoham** [comment]
I commented on the issue. I'll remove the test and let's merge once.

**Review by woodsp-ibm — DISMISSED** [review]

File: releasenotes/notes/remove-snapshot-instruction-aeae8ce4602a7ee2.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+upgrade:
```
Comment (mtreinish) [inline releasenotes/notes/remove-snapshot-instruction-aeae8ce4602a7ee2.yaml:2]: Do we actually need an upgrade release note here? This doesn't really seem like a user facing change to me. It's just internally the aer instruction used under the covers on how you get the expectation value when running on aer right?

Comment (ikkoham) [reply releasenotes/notes/remove-snapshot-instruction-aeae8ce4602a7ee2.yaml:2]: Thanks. Sorry, I thought the release note is required for the internal change... I'll remove it.

**Review by woodsp-ibm — APPROVED** [review]

File: releasenotes/notes/remove-snapshot-instruction-aeae8ce4602a7ee2.yaml (hunk 1 above)
Comment (mtreinish) [reply releasenotes/notes/remove-snapshot-instruction-aeae8ce4602a7ee2.yaml]: Yeah, the release notes are only for user facing changes. It's how we document what's new and/or different for users coming from the previous release. I wrote a little bit about the rationale behind release notes here: https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#release-notes but we probably should expand to make this clearer.

**Pull request merged**